src/modules/claims/claims.service.ts

```diff
@@ -63,6 +63,27 @@ export class ClaimsService {
         this._claimManager = chainConfigs()[chainId].claimManagerAddress;
     }
 
+    /**
+     * A utility function to check the blockchain directly if a DID has a role
+     * TODO: fail if the DID chain ID doesn't match the configured signer network connect
+     * @param did The ethr DID to check
+     * @param role The role to check (the full namespace)
+     * @param version The version to check
+     * @returns true if DID has role at the version. false if not.
+     */
+    async hasOnChainRole(did: string, role: string, version: number): Promise<boolean> {
+        const data = this._claimManagerInterface.encodeFunctionData("hasRole", [
+            addressOf(did),
+            namehash(role),
+            version,
+        ]);
+        const result = await this._signerService.call({
+            to: this._claimManager,
+            data,
+        });
+        return Boolean(Number.parseInt(result));
```

Comment:
Isn't `result` boolean? If yes, then Boolean(paseInt(true)) will return false

Reply:
No, result is a hex string. I added a comment to the code, let me know what you think.

Reply:
Its ok then